torch/csrc/autograd/engine.cpp

```diff
@@ -898,6 +924,11 @@ auto Engine::execute(const edge_list& roots,
       /* depth */ not_reentrant_backward_call ? 0 : total_depth + 1,
       /* cpu_ready_queue */ local_ready_queue);
 
+  // Collects current and default streams for devices where this process has a context,
+  // so GraphTask::exec_post_processing can sync them with leaf_streams.
+  // Should this be inside init_to_execute?
```

Comment:
no, that function (unlike what the name suggests) is not always called and is used for very specific cases only. Here is good.